### PR TITLE
[Form] Added "html5" option to both MoneyType and PercentType

### PR DESCRIPTION
**Pull request by romaricdrigon**
| Q             | A
| ------------- | ---
| Branch?       | master for features
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       |  -
| License       | MIT
| Doc PR        | TBD

Hello,

In the same way that [NumberType](https://symfony.com/doc/current/reference/forms/types/number.html) offers a `html5` option to render a `number` input instead of a `text` input, this PR adds the same option to both `MoneyType` and `PercentType`. Number inputs offer a better UI, especially on mobile (specific keyboard), and they accept extra HTML attributes, such as `max=100`, which are quite useful.

The challenge is that `number` inputs need a "raw" value, nor formatted nor localized.
Format is described here https://www.w3.org/TR/html51/sec-forms.html#date-time-and-number-formats (non-normative, but tested on a few browsers). It matches number formatting for `en` locale (which is great, since it is the one provided by Symfony Intl polyflil), without grouping. 

Implementation was done in a manner similar to `NumberType` for `MoneyType`.
`PercentType` required to modify `PercentToLocalizedStringTransformer` too.

As a bonus, `PercentType` had no tests at all, I added a few.

I wanted to get some feedback on the idea first, remaining steps:
 - update `CHANGELOG.md`
 - update the doc

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
We need to be careful here, type=number is increasingly seen as providing bad UX.
E.g. https://bradfrost.com/blog/post/you-probably-dont-need-input-typenumber/

**Comment by romaricdrigon** [comment]
Thats' a good point, and actually it is specified in the norm that they should not be used for non-sequential inputs (ie., bank account numbers...). Maybe it would be worth a warning in `NumberType` documentation?

Fortunately here with monetary amounts & percentages we should be safe, there are legit use cases.

**Review by javiereguiluz — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
(rebase needed)

**Comment by romaricdrigon** [comment]
Thanks you for the heads up, I rebased code & updated `CHANGELOG.md`
I will work on a PR for documentation tomorrow.

**Comment by romaricdrigon** [comment]
Fabbot and Travis are green, AppVeyor failed because of a quota. If someone can please restart it, it should pass.

File: src/Symfony/Component/Form/CHANGELOG.md
Comment (xabbuh) [inline src/Symfony/Component/Form/CHANGELOG.md:24]: Since 5.1 is in beta now this needs to be added to a new `5.2.0` section.

File: src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php
Comment (xabbuh) [inline src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php:37]: Shouldn't this be the other way around (i.e. `!$options['html5'] ? $options['grouping'] : false`)?

Comment (xabbuh) [reply src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php:37]: On the other hand: We know that `$options['grouping]` is `false` when `html5` is enabled. So we do not need to change anything here at all, do we?

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php
Comment (xabbuh) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php:49]: I think I would rather allow passing a `$locale` and `$grouping` argument like in the `NumberToLocalizedStringTransformer`.

File: src/Symfony/Component/Form/CHANGELOG.md
Comment (romaricdrigon) [reply src/Symfony/Component/Form/CHANGELOG.md:24]: Indeed :S 
I'm updating, thank you for your review.

File: src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php
Comment (romaricdrigon) [reply src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php:37]: That's correct, that's redundant with the normalizer set line 86 (to have a nice error message).

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php
Comment (romaricdrigon) [reply src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php:49]: I have missed feelings about that: HTML5 format happens to be `en` locale without grouping, but it's accidental. I feel like it is correct to expose clearly that we want HTML5-compatible format in the normalizer interface.

**Comment by fabpot** [comment]
What's the status of this PR?

**Comment by romaricdrigon** [comment]
For me, the code part was done. I was waiting for some "formal" approval before doing the doc update.
Everything can be over pretty quickly, I think that addition is still relevant

**Review by nicolas-grekas — APPROVED** [review]
Works for me (I rebased + made some minor tweaks)

**Comment by fabpot** [comment]
Thank you @romaricdrigon.

**Comment by romaricdrigon** [comment]
> Works for me (I rebased + made some minor tweaks)

Thanks you very much @nicolas-grekas for finishing the PR!
I got supra busy early September and less active, but I'm happy to see this ready for 5.2. I will work on the documentation PR.

**Pull request merged**
